webapp/src/pages/[resourceSlug]/task/add.tsx

```diff
@@ -13,17 +13,29 @@ export default function AddTask(props) {
 	const router = useRouter();
 	const { resourceSlug } = router.query;
 	const [state, dispatch] = useState(props);
+	const [cloneState, setCloneState] = useState(null);
 	const [error, setError] = useState();
 	const { tasks, tools, agents } = state;
 
 	async function fetchTaskFormData() {
 		await API.getTasks({ resourceSlug }, dispatch, setError, router);
 	}
 
+	async function fetchEditData(taskId) {
+		await API.getTask({ resourceSlug, taskId }, setCloneState, setError, router);
+	}
+
 	useEffect(() => {
 		fetchTaskFormData();
 	}, [resourceSlug]);
 
+	useEffect(() => {
+		if (typeof location != undefined) {
+			const taskId = new URLSearchParams(location.search).get('taskId');
+			fetchEditData(taskId);
+		}
+	}, []);
+
 	if (tasks == null) {
 		return <Spinner />;
 	}
```

Comment:
Change to use a loading state with initial value set to true, and in a useEffect hook set it to false.

See CrewAppForm and f201781317e69c2b0248838b91096985db98944f